walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -12,12 +12,133 @@
+    Collection<String> optionalAttendeesCollection = request.getOptionalAttendees();
+
+    List<TimeRange> badTimes = new ArrayList<TimeRange>();
+    List<TimeRange> goodTimes = new ArrayList<TimeRange>();
+
+    // get the duration of the event, haven't implemented yet
+    long durationLong = request.getDuration();
+    int duration = (int) durationLong;
+
+    if(duration >= TimeRange.WHOLE_DAY.duration())
+    {
+        return Arrays.asList();
+    }
+
+    Set<String> participants = new HashSet<String>();
+    if(!(request.getOptionalAttendees().isEmpty()))
+    {   
+        for(String attendee: optionalAttendeesCollection)
+        {
+            participants.add(attendee);
+        }
+    }
+
+    for(String attendee: attendeesCollection)
+    {
+        participants.add(attendee);
+    }
+
+
+
+    // before you go any further, I have really bad runtime on here!
+    // sorry! :(
+    //getting each attendee
+
+    for(Event thisMeeting: events) {
+        // get the attendees for this meeting (all meeting participants)
+        Set<String> amp = new HashSet<String>(thisMeeting.getAttendees());
+        // for every attendee in this meeting
+        for(String participant1: amp)
+        {
+            // for every attendee in the meeting we want
+            for(String participant: participants)
+            {
+                // if the participant is in this meeting, add it 
+                // to the bad times we collect
+                if(participant == participant1)
+                {
+                    // if there is a participant who has a meeting that's the whole day,
+                    // then there will be no good meeting times 
+                    if((thisMeeting.getWhen().equals(TimeRange.WHOLE_DAY)))
+                    {
+                        return Arrays.asList();
+                    }
+                    badTimes.add(thisMeeting.getWhen());
+                }
+            }
+        }
+    }
+
+    //if there are no bad times, return the whole day
+    if(badTimes.size() == 0)
+    {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // if there's one event, schedule good times 
+    // before and after
+    if(badTimes.size() == 1)
+    {
+        goodTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, badTimes.get(0).start(), false));
+        goodTimes.add(TimeRange.fromStartEnd(badTimes.get(0).end(), TimeRange.END_OF_DAY, true));
+        return goodTimes;
+    }
+
+    //sort all  meeting (bad) times after we collect them
+    Collections.sort(badTimes, TimeRange.ORDER_BY_START);
+
+    // if we can fit a meeting in before the beginning of the first one
+    // schedule it 
+    if(Math.abs(TimeRange.START_OF_DAY - badTimes.get(0).start()) > duration)
+    {
+        goodTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, badTimes.get(0).start(), false));
+    }
+    
+    /*thought this variable name  was kinda funny!! 
```

Comment:
If you will use // for other comments, please stick with that.

Reply:
Ok